library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -341,7 +364,15 @@ private boolean updateLocked() {
         return isActive;
     }
 
-    private void afterCleanUpDeleteDownload(ContentResolver resolver, DownloadInfo info) {
+    private void updateTotalBytesFor(DownloadInfo info) {
+        if (info.mTotalBytes == -1) {
+            ContentValues values = new ContentValues();
+            info.mTotalBytes = contentLengthFetcher.fetchContentLengthFor(info);
```

Comment:
We proactively fetch the content length for each download in order to later on compute the total size of a batch so that the overall progress is correctly shown.